sbi/inference/snpe/snpe_base.py

```diff
@@ -157,16 +157,21 @@ def __call__(
 
         for round_, num_sims in enumerate(num_sims_per_round):
 
-            # Run simulations for the round.
-            theta, x, prior_mask = self._run_simulations(round_, num_sims)
-            x_shape = x_shape_from_simulation(x)
+            # Run simulations for the round and prepend pre-simulated data.
+            theta, x = self._run_simulations(round_, num_sims)
+            self._append_to_round_bank(theta, x, round_)
+
+            # Load data from most recent round. Will be used for standardizing in the
+            # first round and for shape inference and logging in all other rounds.
+            theta, x, _ = self._get_from_round_bank(round_, exclude_invalid_x)
 
             # First round or if retraining from scratch:
             # Call the `self._build_neural_net` with the rounds' thetas and xs as
-            # arguments, which will build the neural network
+            # arguments, which will build the neural network.
             # This is passed into NeuralPosterior, to create a neural posterior which
             # can `sample()` and `log_prob()`. The network is accessible via `.net`.
             if round_ == 0 or retrain_from_scratch_each_round:
+                x_shape = x_shape_from_simulation(x)
```

Comment:
Would move this down or inside the function call, i.e., `self._handle_x_o_wrt_amortization(x_o, x_shape_from_simulation(x), num_rounds)`

Reply:
It's also needed for posterior initialization, so it has to stay here.